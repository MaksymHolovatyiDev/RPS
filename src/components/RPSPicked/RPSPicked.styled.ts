import { StyleSheet } from 'react-native';
import { styleBase } from '../../stylesBase';

export const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',

    paddingHorizontal: 10,
  },

  itemContainer: {
    justifyContent: 'center',
    alignItems: 'center',
  },

  itemBase: {
    height: 120,
    width: 120,

    backgroundColor: '#14153988',

    borderRadius: 100 / 1,
  },

  text: {
    fontFamily: styleBase.FontBold,
    letterSpacing: 1,

    marginTop: 10,

    textTransform: 'uppercase',
  },
});
