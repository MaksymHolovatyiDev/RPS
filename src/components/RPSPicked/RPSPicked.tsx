import { Text, View } from 'react-native';
import RPSItem from '../RPS/RPSItem';
import { items } from '../RPS/data';
import { styles } from './RPSPicked.styled';
import PlayAgain from '../PllayAgain/PlayAgain';

export default function RPSPicked() {
  return (
    <>
      <View style={styles.container}>
        <View style={styles.itemContainer}>
          <RPSItem data={items[0]} isSelected={true} />
          <Text style={styles.text}>You picked</Text>
        </View>
        <View style={styles.itemContainer}>
          <View style={styles.itemBase}>
            <RPSItem data={items[1]} isSelected={true} />
          </View>
          <Text style={styles.text}>The house picked</Text>
        </View>
      </View>
      <PlayAgain />
    </>
  );
}
