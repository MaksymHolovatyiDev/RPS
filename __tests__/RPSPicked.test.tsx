import {render, screen} from '@testing-library/react-native';
jest.mock('react-native-linear-gradient', () => 'LinearGradient');

import RPSPicked from 'src/components/RPSPicked/RPSPicked';

describe('RPSPicked render', () => {
  it('Should renders correctly', () => {
    jest.useFakeTimers();
    const tree = render(<RPSPicked />).toJSON();

    expect(screen.getByText('You picked')).toBeDefined();
    expect(screen.getByText('The house picked')).toBeDefined();
    expect(screen.getByText('Play again')).toBeDefined();
    expect(tree).toMatchSnapshot();
  });
});
