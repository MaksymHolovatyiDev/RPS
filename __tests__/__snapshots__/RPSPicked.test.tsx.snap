// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RPSPicked render Should renders correctly 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <LinearGradient
          end={
            {
              "x": 0,
              "y": 0.5,
            }
          }
          locations={
            [
              0,
              0.5,
            ]
          }
          start={
            {
              "x": 0,
              "y": 0,
            }
          }
          style={
            {
              "alignItems": "center",
              "borderRadius": 100,
              "height": 120,
              "justifyContent": "center",
              "width": 120,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#fff",
                "borderColor": "#00000033",
                "borderRadius": 100,
                "borderWidth": 3,
                "padding": 20,
              }
            }
          />
        </LinearGradient>
      </View>
      <Text
        style={
          {
            "fontFamily": "BarlowSemiCondensed-Bold",
            "letterSpacing": 1,
            "marginTop": 10,
            "textTransform": "uppercase",
          }
        }
      >
        You picked
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#14153988",
            "borderRadius": 100,
            "height": 120,
            "width": 120,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "opacity": 0,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <LinearGradient
              end={
                {
                  "x": 0,
                  "y": 0.5,
                }
              }
              locations={
                [
                  0,
                  0.5,
                ]
              }
              start={
                {
                  "x": 0,
                  "y": 0,
                }
              }
              style={
                {
                  "alignItems": "center",
                  "borderRadius": 100,
                  "height": 120,
                  "justifyContent": "center",
                  "width": 120,
                }
              }
            >
              <View
                style={
                  {
                    "backgroundColor": "#fff",
                    "borderColor": "#00000033",
                    "borderRadius": 100,
                    "borderWidth": 3,
                    "padding": 20,
                  }
                }
              />
            </LinearGradient>
          </View>
        </View>
      </View>
      <Text
        style={
          {
            "fontFamily": "BarlowSemiCondensed-Bold",
            "letterSpacing": 1,
            "marginTop": 10,
            "textTransform": "uppercase",
          }
        }
      >
        The house picked
      </Text>
    </View>
  </View>,
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "gap": 10,
        "justifyContent": "center",
        "opacity": 0,
        "position": "relative",
        "transform": [
          {
            "translateY": 60,
          },
        ],
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontFamily": "BarlowSemiCondensed-Bold",
          "fontSize": 48,
          "letterSpacing": 2,
          "textTransform": "uppercase",
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 8,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 40,
          "paddingVertical": 14,
          "width": 240,
        }
      }
    >
      <Text
        style={
          {
            "color": "hsl(229, 25%, 31%)",
            "fontFamily": "BarlowSemiCondensed-SemiBold",
            "fontSize": 16,
            "letterSpacing": 2,
            "textTransform": "uppercase",
          }
        }
      >
        Play again
      </Text>
    </View>
  </View>,
]
`;
